robot/framesystem/framesystem.go

```diff
@@ -131,25 +143,38 @@ func (svc *frameSystemService) Reconfigure(ctx context.Context, deps resource.De
 	components := make(map[string]resource.Resource)
 	for name, r := range deps {
 		short := name.ShortName()
-		// is this only for InputEnabled components or everything?
 		if _, present := components[short]; present {
-			return DuplicateResourceShortNameError(short)
+			svc.parts = nil
+			svc.lastReconfigureError = DuplicateResourceShortNameError(short)
+			return svc.lastReconfigureError
 		}
 		components[short] = r
 	}
 	svc.components = components
 
 	fsCfg, err := resource.NativeConfig[*Config](conf)
 	if err != nil {
+		svc.parts = nil
+		svc.lastReconfigureError = err
 		return err
 	}
 
-	sortedParts, err := referenceframe.TopologicallySortParts(fsCfg.Parts)
+	parts, err := svc.ConfigToParts(fsCfg)
 	if err != nil {
+		svc.parts = nil
+		svc.lastReconfigureError = err
```

Comment:
Why do you need to cache the reconfiguration error? Is returning it from `Reconfigure` not enough?

Reply:
The reason for this update is that we often call the function soon to be named `FrameSystemParts` from the motion service (as well as in a couple other places). If we try to do that on a framesystem service that has failed reconfigure we would have to error out but wouldn't know why since we wouldn't be rebuilding it anymore.  Rather than returning something generic I would rather return the reason why it failed reconfigure